media/js/teach/smarton.js

```diff
@@ -0,0 +1,258 @@
+                    $pageFooter.addClass('stuck').append(buttonClose).css({ bottom: '-' + $pageFooter.height() + 'px' }).animate({ bottom: '0' }, 750);
+                    initFooterClose();
+                } else if ((direction === 'up') && (!$body.hasClass('footer-closed'))) {
+                    // slide down when scrolling up, then unstick
+                    $pageFooter.animate({ bottom: '-' + $pageFooter.height() + 'px' }, 500, function() {
+                        $pageFooter.removeClass('stuck').removeAttr('style').find('button.close').remove();
+                    });
+                }
+            }
+        });
+
+        // Unstick the footer when we reach the bottom
+        var unstickyFooter = new Waypoint({
+            element: $('#colophon'),
+            offset: '100%',
+            handler: function(direction) {
+                if ((direction === 'down') && (!$body.hasClass('footer-closed'))) {
+                    $pageFooter.removeClass('stuck').find('button.close').remove();
+                } else if ((direction === 'up') && (!$body.hasClass('footer-closed'))) {
+                    $pageFooter.addClass('stuck').append(buttonClose);
+                    initFooterClose();
+                }
+            }
+        });
+    }
+
+    // Dismiss the sticky footer
+    function initFooterClose() {
+        var footerVisible = $('#footer-cta > .content:visible').data('footer-name');
+        var footerDetails = $('.footer-cta #form-details, .footer-cta .form-details');
+
+        $('.footer-cta .close').on('click', function() {
+            if (tallMode && wideMode) {
+                $pageFooter.animate({ bottom: '-' + $pageFooter.height() + 'px' }, 500, function() {
+                    $pageFooter.removeClass('stuck').removeAttr('style').find('.close').remove();
+                });
+            } else {
+                $pageFooter.removeClass('stuck').removeAttr('style').find('.close').remove();
+            }
+
+            // Close the form if it's open
+            if (footerDetails.is(':visible')) {
+                footerDetails.slideUp('fast');
+            }
+
+            // A class lets us check if the footer was dismissed on purpose
+            // so we don't show it again during scrolling
+            $body.addClass('footer-closed');
+
+            // Track it and record which one was visible
+            window.dataLayer.push({
+                event: 'smarton-interactions',
+                interaction: 'dismiss sticky footer',
+                footer: footerVisible
+            });
+        });
+    }
+
+    // Scroll smoothly to the linked section
+    $document.on('click', '.nav-steps a[href^="#"]', function(e) {
+        e.preventDefault();
+        // Extract the target element's ID from the link's href.
+        var elem = $(this).attr('href').replace(/.*?(#.*)/g, '$1');
+        Mozilla.smoothScroll({
+            top: $(elem).offset().top - $pageHeader.height()
```

Comment:
Hmm yeah, that's not good. Not sure how best to work around it... We could supply a fixed offset that's the estimated height of the bar after it sticks/collapses, but that may be off by a bit if the title wraps to two lines (which it likely will for longer titles, especially when localized). 

Perhaps better to forego the sticky nav entirely on mobile? Then it wouldn't collapse and the height would remain the same, but we lose the title and the indication of which section you're in. Maybe we can live with that. 

I'll be at a conference all day but will have laptop+wifi so I'll come up with something today.